force-app/main/default/classes/ITriggerHandler.cls

```diff
@@ -0,0 +1,31 @@
+public interface ITriggerHandler {
+	void beforeUpdate(
+		Map<Id, SObject> newItemsMap,
+		Map<Id, SObject> oldItemsMap,
+		List<SObject> newItems,
+		List<SObject> oldItems
+	);
+
+	void beforeInsert(List<SObject> newItems);
+
+	void beforeDelete(Map<Id, SObject> oldItemsMap);
+
+	void afterUpdate(
+		Map<Id, SObject> newItemsMap,
```

Comment:
Not needed arguments: List if we have Map. We can simply take List from map if needed by this way:
newItemsMap.values()